### ISSUE TITLE
Add support for ratcheting

### ISSUE THREAD

**Issue by Machine-Maker** [lukebemishprojects/Immaculate#3]
Would be great to mirror the spotless functionality of being able to ratchet changes so as to not have one big "formatting" commit. Basically it just compares the branch with some ref and only runs it on the modified files.

**Comment by lukebemish** [comment]
Immaculate currently has no git integration (and no reason for it in most cases, since gradle's up-to-date checks are quite powerful). That said, I do see the use case for something like what you describe -- I need to think a bit about how to best integrate it into gradle in a nicer way than how spotless approaches it, but I can definitely look into that.

**Comment by Machine-Maker** [comment]
I think you can probably do it already, just by setting the files in a specific workflow to be the modified files. I wrote something (specific to the project I work on) that does it for checkstyle you can see here, no idea if its the optimal way or whatever. https://pastes.dev/fFMzcVlrk9

**Comment by lukebemish** [comment]
Yeah, the basic idea is that I need to run something to determine which files should be considered inputs or not, based on the specified rachet -- probably easiest with a valuesource or the like as a filter? I'll poke it.
